src/main/groovy/wooga/gradle/unity/tasks/GenerateUpmPackage.groovy

```diff
@@ -197,6 +193,89 @@ class GenerateUpmPackage extends Tar implements BaseSpec {
             }
         })
     }
+
+    @Override
+    protected void copy() {
+        if (!packageDirectory.present) {
+            logger.warn(Message.packageDirectoryNotSet.message)
+        }
+
+        project.copy {
+            from(packageDirectory)
+            include(packageManifestFileName)
+            into(temporaryDir)
+        }
+
+        adjustManifestFile()
+
+        from(temporaryDir)
+        from(packageDirectory)
+        super.copy()
+    }
+
+    /**
+     * Modifies the project manifest file (package.json) before it is packaged along with the project)
+     */
+    protected void adjustManifestFile() {
+        def manifest = new File(temporaryDir, packageManifestFileName)
+
+        if (manifest.exists()) {
+
+            // Read
+            def json = new JsonSlurper()
+            Map manifestContent = json.parse(manifest)
+
+            // Apply the default properties
+            if (packageName.present) {
+                manifestContent['name'] = packageName.get()
+            }
+
+            if (archiveVersion.present) {
+                manifestContent['version'] = archiveVersion.get()
+            }
+
+            if (dependencies.present) {
+                manifestContent['dependencies'] = dependencies.get()
+            }
+
+            def _patches = patches.get()
+            manifestContent = merge(manifestContent, _patches)
+
+            // Write back
+            manifest.text = JsonOutput.prettyPrint(JsonOutput.toJson(manifestContent))
+        }
+    }
+
+    /**
+     * Patches the version of a single dependency in the manifest
+     * @param name The name of the package
+     * @param version The version to set
+     */
+    void patchDependency(String name, String version) {
+        def key = "dependencies"
+        if (!patches.get().containsKey(key)) {
+            patches[key] = new HashMap<String, String>()
+        }
+
+        def dependencies = patches[key].get() as Map
+        dependencies[name] = version
+    }
+
+    private static Map merge(Map lhs, Map rhs) {
+
+        // Have to do this since we cannot clone an ImmutableMap
+        def clone = new HashMap()
```

Comment:
Nitpick: Doing `new HashMap(lhs)` doesn't work?

Reply:
That worked as well. Using it.